### PR TITLE
Update k256 to 0.11.0 in casper-types.

### PR DESCRIPTION
**Pull request by zie1ony**
`casper-types` uses `k256` as a dependency. Its current version is `0.7.4` and is 2 years old.
It prevents using `casper-types` with other crates that use newer versions of `k256`.

In addition I had to point `casper-contract` and `casper-types` in `tutorials` to local, which I think is how it should be.

### CODE REVIEW

**Comment by RitaMAllenCA** [comment]
discussion on 2/3/2023 - yes, will consider this strongly, but not for 1.5.  We will discuss when we merge this at a later time.  Marking it for 2.0 right now, but will be considered after 1.5 is released.

**Comment by zie1ony** [comment]
Thank you. It blocks large project on my end. Let's keep it open and I'll remind myself after 1.5 is released.

**Comment by ghost** [comment]
> Its current version is `0.7.4`

@zie1ony , it.s `0.7.2`. Did all tests pass when changing to the  newer version?

> discussion on 2/3/2023 - yes, will consider this strongly, but not for 1.5. We will discuss when we merge this at a later time. Marking it for 2.0 right now, but will be considered after 1.5 is released.

@RitaMAllenCA , what is the rationale here to keep an older version (especially when it's backwards compatible, e.g. no breaking changes)? Changelog indicates many updates of related libraries, and many development on k256.

https://github.com/RustCrypto/elliptic-curves/blob/master/k256/CHANGELOG.md

**Comment by ghost** [comment]
> discussion on 2/3/2023 - yes, will consider this strongly, but not for 1.5. We will discuss when we merge this at a later time. Marking it for 2.0 right now, but will be considered after 1.5 is released.

As a sidenote: this is not an open process. One must be able to follow the rationales of the core-team.

File: smart_contracts/contracts/tutorial/hello-world/Cargo.toml [hunk 1]
```diff
@@ -5,8 +5,8 @@ authors = ["darthsiroftardis <karan@casperlabs.io>", "Michał Papierski <michal@
 edition = "2021"
 
 [dependencies]
-casper-contract = "1.4.4"
-casper-types = "1.5.0"
+casper-contract = { path = "../../../contract" }
+casper-types = { path = "../../../../types" }
```
Comment (ghost) [inline smart_contracts/contracts/tutorial/hello-world/Cargo.toml:9]: why this change? This seems unrelated.

File: types/Cargo.toml [hunk 2]
```diff
@@ -22,7 +22,7 @@ getrandom = { version = "0.2.0", features = ["rdrand"], optional = true }
 hex = { version = "0.4.2", default-features = false, features = ["alloc"] }
 hex_fmt = "0.3.0"
 humantime = { version = "2", optional = true }
-k256 = { version = "0.7.2", default-features = false, features = ["ecdsa", "sha256", "zeroize"] }
+k256 = { version = "0.11.0", default-features = false, features = ["ecdsa", "sha256"] }
```
Comment (ghost) [inline types/Cargo.toml:25]: is the use of "zeroize" mandatory with the new version?

File: types/src/crypto/asymmetric_key.rs [hunk 3]
```diff
@@ -703,7 +703,7 @@ impl From<&PublicKey> for Vec<u8> {
         match public_key {
             PublicKey::System => Vec::new(),
             PublicKey::Ed25519(key) => key.to_bytes().into(),
-            PublicKey::Secp256k1(key) => key.to_bytes().into(),
+            PublicKey::Secp256k1(key) => key.to_bytes().as_slice().into(),
```
Comment (ghost) [inline types/src/crypto/asymmetric_key.rs:706]: "cosmetic" or a mandatory change?

File: types/src/crypto/asymmetric_key/tests.rs [hunk 4]
```diff
@@ -811,7 +811,7 @@ fn should_construct_secp256k1_from_uncompressed_bytes() {
     rng.fill_bytes(&mut secret_key_bytes[..]);
 
     // Construct a secp256k1 secret key and use that to construct a public key.
-    let secp256k1_secret_key = k256::SecretKey::from_bytes(secret_key_bytes).unwrap();
+    let secp256k1_secret_key = k256::SecretKey::from_be_bytes(&secret_key_bytes).unwrap();
```
Comment (ghost) [inline types/src/crypto/asymmetric_key/tests.rs:814]: _be_ = big endian?
mandatory change?

**Comment by ghost** [comment]
>continuous-integration/drone/pr — Build is failing 

Not accessible

**Comment by RitaMAllenCA** [comment]
@Fraser999 @mpapierski @fizyk20 please review and provide feedback.    FYI @rafal-ch

**Comment by Fraser999** [comment]
Updated `k256` to v1.13.1 in #4191.
